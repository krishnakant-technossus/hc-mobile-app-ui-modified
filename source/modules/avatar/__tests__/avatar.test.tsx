// npx jest ./source/modules/avatar/__tests__/avatar.test.ts

import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';

import Avatar from '../avatar';
import ThemeProvider from '../../theming/theme-provider';
import defaultTheme from '../../theming/default-theme';

describe('Avatar', () => {
  describe('Rendering', () => {
    it('renders correctly', () => {
      const wrapper = render(
        <ThemeProvider theme={defaultTheme}>
          <Avatar uri="https://example.com/baby.jpg" />
        </ThemeProvider>,
      );

      const image = wrapper.getByTestId('avatar-image');

      expect(image).toBeDefined();

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Interactions', () => {
    it('should call the onPress function on click', () => {
      const onPressMock = jest.fn();

      const {getByTestId} = render(
        <ThemeProvider theme={defaultTheme}>
          <Avatar uri="https://example.com/baby.jpg" onPress={onPressMock} />
        </ThemeProvider>,
      );

      const button = getByTestId('avatar-button');

      expect(button).toBeDefined();

      fireEvent.press(button);

      expect(onPressMock).toHaveBeenCalledTimes(1);
    });
  });
});
