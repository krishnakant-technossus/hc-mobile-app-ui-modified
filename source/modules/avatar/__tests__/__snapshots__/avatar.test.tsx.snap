// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Rendering renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "height": 33,
      "justifyContent": "center",
      "opacity": 1,
      "overflow": "hidden",
      "width": 33,
    }
  }
  testID="avatar-button"
>
  <Image
    source={
      Object {
        "uri": "https://example.com/baby.jpg",
      }
    }
    style={
      Array [
        Object {
          "borderColor": "#FFFFFF",
          "borderRadius": 33,
          "borderWidth": 1,
          "height": "100%",
          "width": "100%",
        },
      ]
    }
    testID="avatar-image"
  />
</View>
`;
